platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheReport.kt

```diff
@@ -158,9 +163,11 @@ class ConfigurationCacheReport(
             fun ManagedExecutor.shutdownAndAwaitTermination() {
                 shutdown()
                 if (!awaitTermination(30, TimeUnit.SECONDS)) {
+                    val unfinishedTasks = shutdownNow()
                     logger.warn(
                         "Configuration cache report is taking too long to write... "
-                            + "The build might finish before the report has been completely written."
+                            + "The build might finish before the report has been completely written. Unfinished tasks: "
+                            + unfinishedTasks
```

Comment:
💅 I'd suggest to print unfinished tasks separately on a lower log level, (debug or info), as these are going to be very technical (and, likely, not alway actionable for the end user).

Reply:
Done